src/Fantomas/Program.fs

```diff
@@ -271,58 +283,70 @@ let main argv =
 
     let version = results.TryGetResult <@ Arguments.Version @>
 
+    let inline reraiseAsync (e: exn) =
+        let edi = ExceptionDispatchInfo.Capture e
+        edi.Throw()
+
     let fileToFile (force: bool) (inFile: string) (outFile: string) =
-        try
-            printfn $"Processing %s{inFile}"
-            let hasByteOrderMark = hasByteOrderMark inFile
-
-            use buffer =
-                if hasByteOrderMark then
-                    new StreamWriter(
-                        new FileStream(outFile, FileMode.OpenOrCreate, FileAccess.ReadWrite),
-                        Encoding.UTF8
-                    )
+        async {
+            try
+                printfn $"Processing %s{inFile}"
```

Comment:
All of this `printfn` aren't threadsafe and thus we could be printing garbage on the screen. Possibly we should hold off until @dawedawe merges https://github.com/fsprojects/fantomas/pull/2693?

Reply:
Rebased onto the 6.0 branch